src/OpenLoco/src/Vehicles/OrderManager.cpp

```diff
@@ -23,7 +24,23 @@ namespace OpenLoco::Vehicles::OrderManager
     uint8_t* orders() { return getGameState().orders; }
     uint32_t& numOrders() { return getGameState().numOrders; }
 
-    void sub_470795(const uint32_t removeOrderTableOffset, const int16_t sizeOfRemovedOrderTable)
+    void shiftOrdersUp(const uint32_t offsetToShiftTowards, const int16_t sizeToShiftBy)
+    {
+        auto* dest = orders() + offsetToShiftTowards;
+        auto* src = dest + sizeToShiftBy;
+        auto size = S5::Limits::kMaxOrders - offsetToShiftTowards - sizeToShiftBy;
+        std::memmove(dest, src, size);
+    }
+
+    void shiftOrdersDown(const uint32_t offsetToShiftFrom, const int16_t sizeToShiftBy)
+    {
+        auto* src = orders() + offsetToShiftFrom;
+        auto* dest = src - sizeToShiftBy;
+        auto size = S5::Limits::kMaxOrders - offsetToShiftFrom - sizeToShiftBy;
+        std::memmove(dest, src, size);
```

Comment:
I'd prefer to use rotates instead of memove. Then we don't need to mess about with kMaxOrders. Although thinking about it there might be a garbage value moved. std::shift_left would be perfect for this but sadly c++20.